### DISCUSSION TITLE
Can I use Coinbase wallet with BTCPayServer and then convert BTC to USD within Coinbase app?

### DISCUSSION THREAD

**Discussion by olimits7** [btcpayserver/btcpayserver#4047] [Technical Support]
Hi,

I'm in the process of having my developer set up my own BTCPayServer to allow my customers to use BTCPayServer instead of BitPay to make payments in BTC because I've had negative experiences with BitPay.

We're setting up the BTCPayServer using Voltage, I found a YouTube video online that seems to break this out fairly easy and using the BTCPayServer API to work with my website.

However, I'm not sure how the below will work:

- Would I be able to use a Coinbase wallet with my BTCPayServer?
- Would I then be able to convert BTC over to USD within my Coinbase app?
 - If not, how would I be able to convert my BTC over to USD?

Thank you!

**Comment by bolatovumar** [comment]
> Would I be able to use a Coinbase wallet with my BTCPayServer?

No, they are different things. You can send from BTCPay wallet to Coinbase or anywhere else for that matter.

> If not, how would I be able to convert my BTC over to USD?

You can sell BTC using whatever method of selling BTC that you normally use. You can send out BTC from your BTCPay Server to any other wallet. I believe some merchants use [BTCPay Transumeter](https://docs.btcpayserver.org/Transmuter/) to auto-convert BTC to USD by sending to an exchange and selling there.

**Reply by dennisreimann** [reply]
@woutersamaey is working on that front and integrated an API to work with custodian accounts/exchanges. Right now he's working on support for Kraken and I assume the same things can be done for Coinbase as well, at least to the extend that their API allows.

**Reply by woutersamaey** [reply]
Adding Coinbase support (deposit, convert, withdraw) would be like 2 week’s work, but I don’t have an account with them. If this is worth something to you (i.e. you’re willing to pay for my time) and you are okay with me sharing the result with the community, I’m open to discuss it.

**Reply by olimits7** [reply]
Thank you for the replies!  I could also use Kraken as well, do you have an estimate of when this would be possible using Kraken?

**Reply by olimits7** [reply]
> > Would I be able to use a Coinbase wallet with my BTCPayServer?
> 
> No, they are different things. You can send from BTCPay wallet to Coinbase or anywhere else for that matter.
> 

So within my "BTCPay Server" store I would just go under my "**BTC Wallet**" section and then use the "**Send**" button to send to my "**Coinbase**" wallet and then from Coinbase simply sell BTC and buy USD?

Thank you!

**Reply by woutersamaey** [reply]
@olimits7 just check out these PRs. They have been merged into master and some are released:
- https://github.com/btcpayserver/btcpayserver/pull/3923
- https://github.com/btcpayserver/btcpayserver/pull/4024
- https://github.com/btcpayserver/btcpayserver/pull/3978

It already works, but is experimental.

To run this, you need to run my fork of the code.
I'll be preparing proper releases for the general public in the coming weeks & months.

**Reply by olimits7** [reply]
@woutersamaey Ok, great...Ty.  I'll share this with my developer to see if we can get this working with Kraken.

**Reply by olimits7** [reply]
@woutersamaey Just to understand this better, I was looking over the "**Custodian Account**" PRs; so the way this would work is I would be able to add my "**Kraken**" wallet account, for example, as a custodian account and then I would be able to transfer my BTC balance over to my "**Kraken**" wallet?

So my developer would have to integrate the following Greenfield APIs with the "**Deposit**", "**Trade**", "**Withdrawal**" modals because these have not been implemented yet in these PRs?

- **Withdraw:** https://docs.btcpayserver.org/API/Greenfield/v1/#operation/Custodians_WithdrawFromStoreCustodianAccount
  - With this API I would be able to get BTC from my Kraken wallet and put it into my BTCPay Server store wallet?
- **Deposit:** https://docs.btcpayserver.org/API/Greenfield/v1/#operation/Custodians_GetStoreCustodianAccountDepositAddress
  - If I want to deposit BTC from my BTCPay Server wallet to my Kraken wallet I would need to use this API to get my Kraken wallet address?
- **Trade:** https://docs.btcpayserver.org/API/Greenfield/v1/#operation/Custodians_StoreCustodianAccountTradeMarket
  - With this API, I could sell BTC and buy USD?

Thank you!

**Reply by woutersamaey** [reply]
The Greenfield APIs were ready first, and now the UI is being added for easier use.
You will need to install the Kraken plugin which isn't officially released yet, though.

**Reply by olimits7** [reply]
Oh ok, I see…is there an unofficial Kraken plugin I can install or just need to wait until official one is released?

**Reply by woutersamaey** [reply]
Just wait for my release or get it from my personal fork.

**Reply by olimits7** [reply]
@woutersamaey Ok, great…Ty.
